[Docs-2.0] TechnicalOverview.md into FutureOfWasabiWallet.md V2 (#1087)
zkSNACKs/WasabiDoc: docs/building-wasabi/FutureOfWasabiWallet.md

---
{
  "title": "Future Of Wasabi Wallet",

>   "description": "This document aims to describe some ideas for the future of Wasabi Wallet and to explain the reasoning behind going towards these objectives. Plans might change without notice and this only represents our thinking at 2021 December. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop.",

Reviewer: This description is shown in the link preview, it should be highly relevant and short.

```suggestion
  "description": "This document aims to describe some ideas for the future of Wasabi Wallet and to explain the reasoning behind going towards these objectives. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop.",
```

  "sidebarDepth": "1"
}
---

# Ideas And Plans For The Future Of Wasabi Wallet

---

Wasabi Wallet is a privacy focused, non-custodial Bitcoin wallet that is based on the [WabiSabi Framework](https://eprint.iacr.org/2021/206).

While privacy can be achieved today with Wasabi Wallet, Bitcoins mainchain costs, inconvenience and unscalability will inevitably push us to look in to Lightning Network and other 2nd layer solutions.

This document aims to describe some ideas for the future of Wasabi Wallet and to explain the reasoning behind going towards these objectives. 
Plans might change without notice and this only represents our thinking at 2021 December.

> Wasabi does not support and does not plan to support other currencies in the future.

Reviewer: ```suggestion
Wasabi does not support and does not plan to support other currencies in the future, it will always be bitcoin-only.
```

New issues will constantly come up as new users try to use the software.

![Future of Wasabi Wallet in 2021 December](https://user-images.githubusercontent.com/51679301/147867076-bc6e74ed-13f6-455e-8fdb-72ff8e6a03d7.jpg)

https://www.youtube.com/watch?v=aQPNYH3fGAo&ab_channel=WasabiWallet

https://www.youtube.com/watch?v=3etnPUl6HhQ&t=1s&ab_channel=WasabiWallet

https://www.youtube.com/watch?v=9KULN-RkKSE&t=761s&ab_channel=WasabiWallet

[[toc]]

---

## Maintenance

One of the main priorities of Wasabi developers are the stability, performance, code quality, and user experience of Wasabi.
Great care is taken here to build a robust, powerful and simple privacy tool that will attract users, because the more users a network-reliant privacy software has, the better privacy it offers.
In Wasabi, this is most apparent during coinjoins.
The more users that participate in a round, the higher the anonymity set the coinjoin achieves, and the more frequent the rounds will be.

### Coinjoin

Even though Wasabi develoers have done a lot of work to optimize WabiSabi coinjoins privacy and efficiency, the work is still in the beginning. 

> For now, WabiSabi is only enabled for testnet clients, which means there is very little incentive for malicious attackers to reveal if they see a problem in the protocol.

Reviewer 2: ```suggestion
Improvements are still being made.
```

> For now, WabiSabi is only enabled for testnet clients, which means there is very little incentive for malicious attackers to reveal if they see a problem in the protocol.
> Whenever mainchain coinjoin release candidate comes out, we will likely see a lot of new testers with a bunch of issues and improvement ideas.

Reviewer: ```suggestion
```

#### Send in coinjoins

> Even when a user has private coins, being able to send payments in coinjoin is a great chain efficiency and privacy improvement for the sender. 
> Change amount and maybe in some cases the payment amount should be grinded in to next lower standard denomination, making it very difficult to the recipient to figure out anything about the sender. 

Reviewer: ```suggestion
Being able to send payments in coinjoin is a great blockspace efficiency and privacy improvement for both the sender and the receiver.
Change amount and maybe in some cases the payment amount should be split into standard denominations.
With Wabisabi pay to endpoint users can send each other money without the sender knowing the address of the receiver, or the receiver knowing the inputs of the sender.
This also enables anonymous transaction batching.
```

GroupSend: [https://github.com/zkSNACKs/Meta/issues/29](https://github.com/zkSNACKs/Meta/issues/29)

#### Round Interoperability

> Because Bitcoin blocks have size limit and every input and output in a coinjoin has weight, it is not possible to add thousands of transactions to a coinjoin round. 

Reviewer: ```suggestion
Because bitcoin blocks have a size limit, it is not possible to add thousands of inputs and outputs to a coinjoin transaction.
```

To enable exponentially groving privacy we need some of the users to remix, making the coinjoin rounds interconnected, resulting in anonymity set growth somewhere between the scales of addition and multiplication for all the rounds participants. 
If that doesn't happen enough by default, the coordinator has ways to incentivice users by for example, giving fee discounts for remixed inputs.

### Wallet

Wasabi Wallet devs should keep up with the latest changes in other Bitcoin wallets and implement the most important features to Wasabi, while also developing new features. 

#### Coin Control

Wasabi Wallet was one of the first wallets that implemented coin control, that demonstrated the UTXOs to the user, giving them a lot of options on how to control their funds. 
Later developers realized that especially new users found this inconvenient, scary and confusing, so they decided to design a new wallet that uses Automated Coin Selection, enabling them to remove the old coin control whenever there is enough private coins to make a payment.

> Currently Wasabi only shows cluster control window when user doesn't have enough of it's balance as private.

Reviewer 2: ```suggestion
Currently Wasabi only shows the cluster control window when the user doesn't have enough of it's balance as private.
```

NOTE: Developers might enable Coin Control for the advanced users in the future.

#### Simple Send Improvements

For non-coinjoined coins, improving simple send using WabiSabi is also an interesting topic.
These do not even have to disrupt the current user workflow, they can mostly "just happen" in the background.
Send in coinjoin [https://github.com/zkSNACKs/Meta/issues/6](https://github.com/zkSNACKs/Meta/issues/6), [https://github.com/nopara73/ZeroLink/issues/75](https://github.com/nopara73/ZeroLink/issues/75) has great potential.

> #### P2M

Reviewer: ```suggestion
#### Transaction batching
```

> Wasabi should enable users to make payments to multiple addresses in the same transaction, in a convenient and privacy presenving way. 

Reviewer 2: ```suggestion
Wasabi should enable users to make payments to multiple addresses in the same transaction, in a convenient and privacy preserving way. 
```

> This would save users some sats in transaction fees but it comes with the cost of all recipients being able to see the other outputs in the transaction. 

Reviewer: ```suggestion
This would save users sats in mining fees but it comes with the cost of all recipients being able to see the other outputs in the transaction.
```

> If the transaction doesn't have a lot of them, this might be a privacy issue. 

Reviewer: ```suggestion
```

> Some users might be ok with that but combining P2M with Send in Coinjoins would definitely make it more private for everyone.

Reviewer: ```suggestion
Some users might be ok with that but making batched payments inside coinjoin would definitely make it more private for everyone.
```

Pay to Many: https://github.com/zkSNACKs/WalletWasabi/issues/733

#### Advanced RBF

> 

Reviewer: ```suggestion

This allows users to speed up the confirmation time of transactions, and even cancel unconfirmed payments.

Fee bumping coinjoin transactions is expensive, but possible for any output.
A service provider can spend his coinjoin output in an rbf transaction, and then fee bump that spending transaction whenever users pay him off-chain.

```

Advanced Replace-By-Fee: [https://github.com/zkSNACKs/Meta/issues/15](https://github.com/zkSNACKs/Meta/issues/15)

#### Sweep Private Key

This could be helpful especially for the holders of physical coins, paper wallets or OpenDimes.
Sweep private key: [https://github.com/zkSNACKs/WalletWasabi/issues/486](https://github.com/zkSNACKs/WalletWasabi/issues/486).
Since Wasabi doesn't support old address formats, a single backend query could establish wallet balance only once, and then immediately sweep the money to a bech32 wallet-managed address: [https://github.com/zkSNACKs/Meta/issues/34](https://github.com/zkSNACKs/Meta/issues/34).

#### Privacy Calculation

If in every coinjoin round, more than half of the inputs are remixes, it not only results in perfect mixes for those inputs, but also results in anonymity set growth somewhere between the scales of addition and multiplication, instead of simple addition as the ZeroLink paper anticipated.
Whether the anonymity set gain is closer to addition or multiplication depends on how other users behave.
Right now, Wasabi simply counts the worst case scenario: as addition.

> However, this is happening in a low Bitcoin fee environment, so this is not to be taken for granted in the future.

Reviewer: ```suggestion
However, this is happening in a low transaction fee environment, so this is not to be taken for granted in the future.
```

Additional measures are necessary.
Accurate Anonymity Set Calculation: [https://github.com/zkSNACKs/Meta/issues/35](https://github.com/zkSNACKs/Meta/issues/35)

#### Bitcoin Address Format

> Currently Wasabi Wallet uses bech32 addresses (bc1q...) only but developers are working on enabling P2TR (bc1p...).

Reviewer: ```suggestion
Currently Wasabi Wallet uses only segwit v0 addresses (bc1q...) but developers are working on enabling taproot addresses (bc1p...).
```

Older address formats (1... and 3...) are not checked against Golomb-Rice filters.
Instead, a single backend query could establish its balance only once, and then immediately sweep the money to a bech32 wallet-managed address: [https://github.com/zkSNACKs/Meta/issues/34](https://github.com/zkSNACKs/Meta/issues/34).

### User Experience

> Improving the user friendliness of the software, the accuracy of coin awareness, and presenting what happens on the blockchain can be beneficial.

Reviewer: ```suggestion
Improving the user friendliness of the software, the accuracy of coin awareness, and presenting what happens on the blockchain can be beneficial.

The default UX should be very simple and intuitive, with a lot of concepts abstracted and under the hood.
But at each point the user can opt-in to more insight and control.
```

#### QR-Scanner

QR code reader works in Windows. WIP for Linux and Mac. // Link to Pindurs PRs and issues.

## Marketing

While education, content creation and marketing have little place in a technical documentation, they are still important parts of the big picture.

### Education

Education is a vitally important aspect of Bitcoin privacy in general and Wasabi specifically.
Through education, Wasabi can obtain new users.
The more Wasabi users there are, the better their privacy.
Advancing this issue can take various, often opportunistic forms.

- Add Infographics to software [https://github.com/zkSNACKs/Meta/issues/32](https://github.com/zkSNACKs/Meta/issues/32)
- Implement Bitcoin Academy [https://github.com/zkSNACKs/Meta/issues/33](https://github.com/zkSNACKs/Meta/issues/33)

### Events

> Wasabi Wallet contributors are encouraged to participate and maybe even create Bitcoin and privacy related events, to make sure they stay up to date about the latest research and features.

Reviewer: ```suggestion
Wasabi Wallet contributors are encouraged to participate and create bitcoin and privacy related events, to make sure they stay up to date about the latest research and features.
```

Therefore, Wasabi should also pay attention to fields that help to increase the number of Wasabi users, bringing greater privacy for everyone.

### Mobile Wallet

In theory, Wasabi could support smart phones (Android, iOS).
In practice, these platforms and their tools are not mature enough just yet.
The concept of network-analysis resistant smartphone wallets is not yet proven.

If we attempted to port Wasabi's code today, the wallet would use too much storage space, battery and network.
However, technology is improving quickly, thus, timing has special importance in this matter. [https://github.com/zkSNACKs/Meta/issues/9](https://github.com/zkSNACKs/Meta/issues/9)

### Localization

Since most of the world does not speak English, localization ([https://github.com/zkSNACKs/Meta/issues/22](https://github.com/zkSNACKs/Meta/issues/22)) of Wasabi is something to consider after Mobile Wallet.

### Tooling

Developers are aiming to modularize WabiSabi coinjoin to enable other projects, wallets and companies to implement it.

#### RPC

> Wasabi offers a daemon and an RPC interface.

Reviewer: ```suggestion
Wasabi currently offers a rudimentary RPC interface.
This should be improved in architecture and functionality.
```

> Full information can be found [here](/using-wasabi/Daemon.md) and [here](/using-wasabi/RPC.md).

Reviewer: ```suggestion
Full information can be found [here](/using-wasabi/RPC.md).
```

#### Daemon/API

> Wasabi offers a daemon and an RPC interface.
> Full information can be found [here](/using-wasabi/Daemon.md) and [here](/using-wasabi/RPC.md).

Reviewer: ```suggestion
Wasabi should be able to start and run in a headless daemon mode with full wallet functionality via the RPC server.
```

#### Documentation

> Working on it! 

Reviewer: ```suggestion
There should be a comprehensive documentation with guides of how to use the software, and more information about how it works under the hood.
```

> #### WAT?

Reviewer 2: ?

> #### WAT?

Reviewer: ```suggestion
```

#### Tutorials

https://www.youtube.com/watch?v=bQKU2khua-8&t=442s&ab_channel=WasabiWallet

#### Companies

Wasabi developers are aiming to make revenue sharing possible with other companies. 
By them integrating WabiSabi, they receive sats everytime their users participates in a coinjoin and pays coordinator fees.

#### Integration

> Developers are aiming to make revenue sharing possible with other companies. 
> By them integrating WabiSabi, they receive sats everytime their users participates in a coinjoin and pays coordinator fees.

Reviewer: ```suggestion
```

It would be also beneficial and not difficult to integrate Wasabi into existing .NET softwares directly, like [BTCPay](https://github.com/btcpayserver/btcpayserver).

> 

Reviewer: ```suggestion
But WabiSabi should be easily usable by software written in other languages.

```

#### Libraries

#### Bitcoin Full Node integration

Improve full node integration to minimize reliance to backend.

### Hardware Wallets

Wasabi uses the [Bitcoin Core Hardware Wallet Interface [HWI]](https://github.com/bitcoin-core/HWI) ([PR #1341](https://github.com/zkSNACKs/WalletWasabi/pull/1341) & [PR #1905](https://github.com/zkSNACKs/WalletWasabi/pull/1905)) however in this mode coinjoining is currently not possible. We should fix this.

## Innovation

> Wasabi should continue developing and implementing new in-house features that has become industry standards during the years, like discreet mode, coin control and coinjoins.

Reviewer: ```suggestion
Wasabi should continue researching and implementing new concepts and features.
```

Additional measures are necessary.
Accurate Anonymity Set Calculation: [https://github.com/zkSNACKs/Meta/issues/35](https://github.com/zkSNACKs/Meta/issues/35)

#### Bitcoin Address Format

Currently Wasabi Wallet uses only segwit v0 addresses (bc1q...) but developers are working on enabling taproot addresses (bc1p...).

Older address formats (1... and 3...) are not checked against Golomb-Rice filters.
Instead, a single backend query could establish its balance only once, and then immediately sweep the money to a bech32 wallet-managed address: [https://github.com/zkSNACKs/Meta/issues/34](https://github.com/zkSNACKs/Meta/issues/34).

### User Experience

Improving the user friendliness of the software, the accuracy of coin awareness, and presenting what happens on the blockchain can be beneficial.

The default UX should be very simple and intuitive, with a lot of concepts abstracted and under the hood.
But at each point the user can opt-in to more insight and control.

#### QR-Scanner

QR code reader works in Windows. WIP for Linux and Mac. // Link to Pindurs PRs and issues.

> ## Marketing
> 
> While education, content creation and marketing have little place in a technical documentation, they are still important parts of the big picture.
> 
> ### Education
> 
> Education is a vitally important aspect of Bitcoin privacy in general and Wasabi specifically.
> Through education, Wasabi can obtain new users.
> The more Wasabi users there are, the better their privacy.
> Advancing this issue can take various, often opportunistic forms.
> 
> - Add Infographics to software [https://github.com/zkSNACKs/Meta/issues/32](https://github.com/zkSNACKs/Meta/issues/32)
> - Implement Bitcoin Academy [https://github.com/zkSNACKs/Meta/issues/33](https://github.com/zkSNACKs/Meta/issues/33)
> 
> ### Events
> 
> Wasabi Wallet contributors are encouraged to participate and create bitcoin and privacy related events, to make sure they stay up to date about the latest research and features.

Reviewer: ```suggestion
```

### Contribution games

The company behind Wasabi Wallet (zkSNACKs Ltd.) have been doing contribution games for years and they have proven to be very beneficial for the contributors, for developers and for the users that end up enjoying the improved wallet.
Current ones can be found in the [blog](https://blog.wasabiwallet.io/coming-soon-contribution-games/)

> ### Merchandise Shop
> 
> Besides of the online Bitcoin privacy that Wasabi Wallet offers, we could spin up a merchandise store that offers shirts, mugs, hats and some of the best meat-space privacy tools like curtains and underware.

Reviewer: ```suggestion
```

---
{
  "title": "Future Of Wasabi Wallet",
  "description": "This document aims to describe some ideas for the future of Wasabi Wallet and to explain the reasoning behind going towards these objectives. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop.",
  "sidebarDepth": "1"
}
---

# Ideas And Plans For The Future Of Wasabi Wallet

---

Wasabi Wallet is a privacy focused, non-custodial Bitcoin wallet that is based on the [WabiSabi Framework](https://eprint.iacr.org/2021/206).

While privacy can be achieved today with Wasabi Wallet, Bitcoins mainchain costs, inconvenience and unscalability will inevitably push us to look in to Lightning Network and other 2nd layer solutions.

This document aims to describe some ideas for the future of Wasabi Wallet and to explain the reasoning behind going towards these objectives. 
Plans might change without notice and this only represents our thinking at 2021 December.

> Wasabi does not support and does not plan to support other currencies in the future, it will always be bitcoin-only.

Reviewer 3: This is not true. Wasabi is bitcoin-only, because Bitcoin is going to be the most likely winner of the currency wars. We should certainly not be religious about it and if Bitcoin is losing its market share, this position must be reconsidered.

https://github.com/zkSNACKs/Meta#bitcoin-only

> Wasabi does not support and does not plan to support other currencies in the future, it will always be bitcoin-only.

Reviewer: ```suggestion
```

Wasabi Wallet is software, therefore it must not neglect general software quality issues.
The better software Wasabi is, the more users it will retain.

Wasabi is also a Bitcoin wallet, therefore it must improve general Bitcoin wallet related features as well.
New issues will constantly come up as new users try to use the software.

> ![Future of Wasabi Wallet in 2021 December](https://user-images.githubusercontent.com/51679301/147867076-bc6e74ed-13f6-455e-8fdb-72ff8e6a03d7.jpg)

Reviewer 3: Please censor the companies in there. They may not want this to be public and we should respect that.

https://www.youtube.com/watch?v=aQPNYH3fGAo&ab_channel=WasabiWallet

https://www.youtube.com/watch?v=3etnPUl6HhQ&t=1s&ab_channel=WasabiWallet

https://www.youtube.com/watch?v=9KULN-RkKSE&t=761s&ab_channel=WasabiWallet

[[toc]]

---

## Maintenance

One of the main priorities of Wasabi developers are the stability, performance, code quality, and user experience of Wasabi.
Great care is taken here to build a robust, powerful and simple privacy tool that will attract users, because the more users a network-reliant privacy software has, the better privacy it offers.
In Wasabi, this is most apparent during coinjoins.
The more users that participate in a round, the higher the anonymity set the coinjoin achieves, and the more frequent the rounds will be.

### Coinjoin

Even though Wasabi develoers have done a lot of work to optimize WabiSabi coinjoins privacy and efficiency, the work is still in the beginning. 

#### Send in coinjoins

> Being able to send payments in coinjoin is a great blockspace efficiency and privacy improvement for both the sender and the receiver.

Reviewer 3: The privacy improvement is none to little.

Reviewer 3: Ok, well, this is convincing: https://github.com/zkSNACKs/WasabiDoc/pull/1087/files#r917386032

Reviewer: It is a substantial improvement, the receiver does not know which inputs or other outputs the sender has.

Reviewer 3: They don't know already. Users coinjoin before send and after send.

Change amount and maybe in some cases the payment amount should be split into standard denominations.

> With Wabisabi pay to endpoint users can send each other money without the sender knowing the address of the receiver, or the receiver knowing the inputs of the sender.

Reviewer 3: Hmm. I wonder if this is feasible at all.

This also enables anonymous transaction batching.
GroupSend: [https://github.com/zkSNACKs/Meta/issues/29](https://github.com/zkSNACKs/Meta/issues/29)

#### Round Interoperability

Because bitcoin blocks have a size limit, it is not possible to add thousands of inputs and outputs to a coinjoin transaction.
To enable exponentially groving privacy we need some of the users to remix, making the coinjoin rounds interconnected, resulting in anonymity set growth somewhere between the scales of addition and multiplication for all the rounds participants. 

> If that doesn't happen enough by default, the coordinator has ways to incentivize users, for example by giving fee discounts for remixed inputs.

Reviewer 3: This happens by default, so this section can be removed.

> If that doesn't happen enough by default, the coordinator has ways to incentivize users, for example by giving fee discounts for remixed inputs.

Reviewer: ```suggestion
```

### Wallet

Wasabi Wallet devs should keep up with the latest changes in other Bitcoin wallets and implement the most important features to Wasabi, while also developing new features. 

> #### Coin Control

Reviewer 3: Coin control is obsolete. Remove this.

Reviewer: Only obsolete in the default UX, not in the wallet as a whole though.

Reviewer 3: It's obsolete as a privacy technology.

Wasabi Wallet was one of the first wallets that implemented coin control, that demonstrated the UTXOs to the user, giving them a lot of options on how to control their funds. 
Later developers realized that especially new users found this inconvenient, scary and confusing, so they decided to design a new wallet that uses Automated Coin Selection, enabling them to remove the old coin control whenever there is enough private coins to make a payment.
Currently Wasabi only shows the cluster control window when the user doesn't have enough of it's balance as private.
NOTE: Developers might enable Coin Control for the advanced users in the future.

> #### Simple Send Improvements

Reviewer 3: Meh

For non-coinjoined coins, improving simple send using WabiSabi is also an interesting topic.
These do not even have to disrupt the current user workflow, they can mostly "just happen" in the background.
Send in coinjoin [https://github.com/zkSNACKs/Meta/issues/6](https://github.com/zkSNACKs/Meta/issues/6), [https://github.com/nopara73/ZeroLink/issues/75](https://github.com/nopara73/ZeroLink/issues/75) has great potential.

> #### P2M

Reviewer 3: It's meh too. Not because I don't want this, but because it's like a miniscule feature just like the previous ones. This document should focus on higher level and more useful things, rather than all kinds of little things.

Wasabi should enable users to make payments to multiple addresses in the same transaction, in a convenient and privacy preserving way. 
This would save users sats in mining fees but it comes with the cost of all recipients being able to see the other outputs in the transaction.
Some users might be ok with that but making batched payments inside coinjoin would definitely make it more private for everyone.
Pay to Many: https://github.com/zkSNACKs/WalletWasabi/issues/733

#### Advanced RBF

This allows users to speed up the confirmation time of transactions, and even cancel unconfirmed payments.

Fee bumping coinjoin transactions is expensive, but possible for any output.
A service provider can spend his coinjoin output in an rbf transaction, and then fee bump that spending transaction whenever users pay him off-chain.
Advanced Replace-By-Fee: [https://github.com/zkSNACKs/Meta/issues/15](https://github.com/zkSNACKs/Meta/issues/15)

> #### Sweep Private Key

Reviewer 3: Another meh. I want it, but it's super uninportant.

Additional measures are necessary.
Accurate Anonymity Set Calculation: [https://github.com/zkSNACKs/Meta/issues/35](https://github.com/zkSNACKs/Meta/issues/35)

#### Bitcoin Address Format

Currently Wasabi Wallet uses only segwit v0 addresses (bc1q...) but developers are working on enabling taproot addresses (bc1p...).

Older address formats (1... and 3...) are not checked against Golomb-Rice filters.
Instead, a single backend query could establish its balance only once, and then immediately sweep the money to a bech32 wallet-managed address: [https://github.com/zkSNACKs/Meta/issues/34](https://github.com/zkSNACKs/Meta/issues/34).

### User Experience

Improving the user friendliness of the software, the accuracy of coin awareness, and presenting what happens on the blockchain can be beneficial.

The default UX should be very simple and intuitive, with a lot of concepts abstracted and under the hood.
But at each point the user can opt-in to more insight and control.

#### QR-Scanner

QR code reader works in Windows. WIP for Linux and Mac. // Link to Pindurs PRs and issues.

> ## Marketing

Reviewer 3: You can remove this section. It's has no place in "technical overview"

### Merchandise Shop

Besides of the online Bitcoin privacy that Wasabi Wallet offers, we could spin up a merchandise store that offers shirts, mugs, hats and some of the best meat-space privacy tools like curtains and underware.

## Scaling

For now the focus has been on a robust and easy to use non-custodial Bitcoin desktop wallet with built-in coinjoins.
While Bitcoin continues to grow, we have to also spread out the privacy that WabiSabi provides to other devices, platforms and wallets.
In the paper [Anonymity Loves Company: Usability and the Network Effect](https://www.freehaven.net/anonbib/cache/usability:weis2006.pdf) the authors note:

> We show that in anonymizing networks, even if you were smart enough and had enough time to use every system perfectly, you would nevertheless be right to choose your system based in part on its usability for other users.
Therefore, Wasabi should also pay attention to fields that help to increase the number of Wasabi users, bringing greater privacy for everyone.

### Mobile Wallet

In theory, Wasabi could support smart phones (Android, iOS).
In practice, these platforms and their tools are not mature enough just yet.

> The concept of network-analysis resistant smartphone wallets is not yet proven.

Reviewer 3: It is proven already. See Breez wallet.

> If we attempted to port Wasabi's code today, the wallet would use too much storage space, battery and network.

Reviewer 3: I don't like this. It's not really about the timing, it's about how much work it is :)

### Tooling

Developers are aiming to modularize WabiSabi coinjoin to enable other projects, wallets and companies to implement it.

#### RPC

Wasabi currently offers a rudimentary RPC interface.
This should be improved in architecture and functionality.
Full information can be found [here](/using-wasabi/RPC.md).

#### Daemon/API

Wasabi should be able to start and run in a headless daemon mode with full wallet functionality via the RPC server.

#### Documentation

There should be a comprehensive documentation with guides of how to use the software, and more information about how it works under the hood.

#### Tutorials

https://www.youtube.com/watch?v=bQKU2khua-8&t=442s&ab_channel=WasabiWallet

#### Companies

Wasabi developers are aiming to make revenue sharing possible with other companies. 
By them integrating WabiSabi, they receive sats everytime their users participates in a coinjoin and pays coordinator fees.

#### Integration

It would be also beneficial and not difficult to integrate Wasabi into existing .NET softwares directly, like [BTCPay](https://github.com/btcpayserver/btcpayserver).
But WabiSabi should be easily usable by software written in other languages.

> #### Libraries

Reviewer 3: It means software libraries, not real world ones :D
